### DISCUSSION TITLE
Docker: Invalid Reference Format - trying to execute Docker run command

### DISCUSSION THREAD

**Discussion by begunfx** [haugene/docker-transmission-openvpn#2547] [Q&A] [answered]
I'm having trouble trying to build the docker container for this project.  I think it might be because I have special characters in my PIA password.  What is the proper way to handle the special characters when using the docker run command?  I tried escaping them with a backslash but no luck.  Any help would be greatly appreciated.    My run command is below (username and password redacted).

`docker run --cap-add=NET_ADMIN -d \
--name=transmission \
-v /volume1/docker/transmission/data:/data \
-v /volume1/docker/transmission/config:/config \
-e OPENVPN_PROVIDER=PIA \
-e OPENVPN_CONFIG=us_california, us_las_vegas, us_seattle, us_west \
-e OPENVPN_USERNAME=xxxxx \
-e OPENVPN_PASSWORD=xxxxx \
-e LOCAL_NETWORK=192.168.1.0/24 \
-e TZ=America/Los_Angeles \
--log-driver json-file \
--log-opt max-size=10m \
--restart always \
-p 9091:9091 \
haugene/transmission-openvpn`

**Comment by pkishino** [comment]
you can try this:
https://haugene.github.io/docker-transmission-openvpn/faq/#send_username_and_password_via_a_file
https://haugene.github.io/docker-transmission-openvpn/config-options/#required_environment_options

**Comment by begunfx** [comment]
Thanks for the response @pkishino.  I moved the credentials to a file in /config and set the env variable as described, but I'm still getting the following error:

Current status: 125 (Interrupted)
Standard output/error:
docker: invalid reference format.

Maybe it's not the special characters in the password field?
I can try setting PUID and PGID, but I'm not sure that will make much of a difference.  Any other suggestions?  I'm running on a Synology DSM 7+.  Thanks!

**Comment by begunfx** [comment, marked as answer]
Okay  I figured it out.  It was the spaces between the different PIA servers that kept throwing that error.   On a side note:  I switched my credentials to a file, but was getting a special character error using this example from your link:


$ docker run --cap-add=NET_ADMIN -d \
              -v /your/storage/path/:/data \
              -v ./openvpn-credentials.txt:/config/openvpn-credentials.txt \
              -e OPENVPN_PROVIDER=PIA \
              -e OPENVPN_CONFIG=france \
              -e OPENVPN_USERNAME=**None** \
              -e OPENVPN_PASSWORD=**None** \
              -e LOCAL_NETWORK=192.168.0.0/16 \
              --log-driver json-file \
              --log-opt max-size=10m \
              -p 9091:9091 \
              haugene/transmission-openvpn

for the credentials, I was able to solve it by removing the relative path "./openvpn-credentials.txt" and using the explicit path: "/your/storage/path/config/openvpn-credentials.txt".

**Reply by pkishino** [reply]
Ah, yes, for volume mounts it needs to be absolute.. for configs, no space should works right?

**Reply by begunfx** [reply]
Correct.  I had to change the credentials to be absolute and that fixed it.  :-)
